docs/proposed/psa-driver-interface.md

```diff
@@ -536,6 +549,21 @@ If the driver has an [`"allocate_key"` entry point](#key-management-in-a-secure-
 
 TODO: derivation, copy
 
+#### Key validation and size on import
+
+The `"import_key"` entry point must validate the key so that if a key is imported successfully, permitted operations on the key will succeed if the input data is valid and enough resources are available. For key types that are defined in the PSA Cryptography specification, opaque drivers must guarantee the properties that transparent drivers guarantee if [`"validate_key"`](#key-validation-with-transparent-drivers) succeeds.
```

Comment:
Why not as in the transparent case have a `validate_key` entry point to do the validation ? If there is no reason to do it differently, probably better to do things the same way for transparent and opaque drivers.

Reply:
Validation requires parsing the key data and making calculations on the in-memory representation. Import requires parsing the key data, probably doing the validation (because the secure element couldn't trust the core to have done it), and writing an internal serializable representation. Having a separate entry point for validation would require duplicating some of the work.

`validate` for transparent keys is really `import`, but because the serializable representation for transparent keys is identical to the external representation, it's an import that doesn't have any output for key material, just the possibility of erroring out (and also reporting missing metadata, specifically the key size).